discovery_engine_core/core/src/document.rs

```diff
@@ -54,6 +54,12 @@ impl Id {
     pub(crate) fn new() -> Self {
         Self(Uuid::new_v4())
     }
+
+    #[cfg(feature = "storage")]
+    #[cfg(test)]
```

Comment:
:art: the attributes could be logically combined
```suggestion
    #[cfg(all(feature = "storage", test))]
```